File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        times = 0
        pa, pb = points[0]
        for ca, cb in points[1:]:
            times += max(abs(pa - ca), abs(cb - pb))
            pa, pb = ca, cb
        
        return times
            

File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.go
func minTimeToVisitAllPoints(points [][]int) int {
    res := 0
    for i := 1; i < len(points); i++ {
        dx := abs(points[i - 1][0] - points[i][0])
        dy := abs(points[i - 1][1] - points[i][1])
        res += dx
        if dx < dy {
            res += (dy - dx)
        }
    }
    return res
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}